### PR TITLE
Integrate cpm

### PR DESCRIPTION
**Pull request by mgerhold**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,3 +3,40 @@ CMAKE_MINIMUM_REQUIRED(VERSION 3.26)
 project(sillysprites)
 
 add_executable(sandbox src/sandbox/main.cpp)
+
+include(${CMAKE_SOURCE_DIR}/cmake/CPM.cmake)
+
+TARGET_INCLUDE_DIRECTORIES(sandbox PUBLIC ${CMAKE_SOURCE_DIR}/vendor)
```
Comment (PurpurTentakel97) [inline CMakeLists.txt:9]: random caps

File: src/sandbox/main.cpp [hunk 2]
```diff
@@ -1,5 +1,35 @@
...
 int main() {
-    std::cout << "Hello, world!\n";
+    spdlog::info("Hello, world!");
+    auto vec = tl::optional<glm::vec3>{
+        glm::vec3{1.0, 2.0, 3.0}
+    };
+    if (vec.has_value()) {
+        spdlog::info("{},{},{}", vec->x, vec->y, vec->z);
+    }
+    const auto value = gsl::narrow_cast<float>(3.14);
```
Comment (PurpurTentakel97) [inline src/sandbox/main.cpp:28]: wrong const

File: src/sandbox/main.cpp [hunk 3]
```diff
@@ -1,5 +1,35 @@
...
+    spdlog::info("{}", value);
+    spdlog::info("this is just a test!");
+
+    for (const auto color : magic_enum::enum_values<Color>()) {
```
Comment (PurpurTentakel97) [inline src/sandbox/main.cpp:32]: wrong const

**Review by PurpurTentakel97 — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (mgerhold) [reply CMakeLists.txt:9]: fixed

File: src/sandbox/main.cpp (hunk 2 above)
Comment (mgerhold) [reply src/sandbox/main.cpp:28]: fixed

File: src/sandbox/main.cpp (hunk 3 above)
Comment (mgerhold) [reply src/sandbox/main.cpp:32]: fixed

**Review by PurpurTentakel97 — APPROVED** [review]

**Pull request merged**